ios/REANodesManager.m

```diff
@@ -552,4 +585,31 @@ - (NSString *)obtainProp:(nonnull NSNumber *)viewTag propName:(nonnull NSString
   return result;
 }
 
+- (void)tryToFlushUpdateBuffer
+{
+  if (_componentUpdateBuffer.count == 0) {
+    return;
+  }
+
+  __weak typeof(self) weakSelf = self;
+  RCTExecuteOnMainQueue(^void() {
```

Comment:
As discussed offline, we need to make sure the updates are flushed in the same UI loop in which the component is mounted. There are two problems here:
1) testing for _componentUpdateBuffer is not thread safe as the prop is not synchronized and being updated from other threads
2) AFAIK RCTExecuteOnMainQueue will schedule execution for the following frame and not for the current one. I think you may try using addUIBlock from UIManager, but maybe there are also other ways

Reply:
1. I added synchronization for `_componentUpdateBuffer`.

2. I used the method `addUIBlock` and according to https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/React/Modules/RCTUIManager.m#L1171 even if we use it inside of`uiManagerWillPerformMounting`, the updates should be applied during the same frame.